app/services/katello/pulp3/repository/generic.rb

```diff
@@ -30,6 +36,59 @@ def self.api(smart_proxy, repo)
           api_class = RepositoryTypeManager.find_by(:pulp3_service_class, self).pulp3_api_class
           api_class ? api_class.new(smart_proxy, repo) : Katello::Pulp3::Api::Core.new(smart_proxy)
         end
+
+        def create_distribution(path)
+          distribution_data = api.class.distribution_class(repo.repository_type).new(secure_distribution_options(path))
+          api.distributions_api.create(distribution_data)
+        end
+
+        def create_remote
+          remote_file_data = api.class.remote_class(repo.repository_type).new(remote_options)
+          response = api.remotes_api.create(remote_file_data)
+          repo.update!(:remote_href => response.pulp_href)
+        end
+
+        def refresh_distributions
```

Comment:
I still wonder if we could reuse more `refresh_distributions` code from ::Katello::Pulp3::Repository.  I'll leave that to the next reviewer's opinion.

Reply:
oh sorry i missed this comment, y eah this seem nearly identical to the one there, are there any differences?  if not, let just delete this method.  If there are differences, maybe a small followup pr is fine

Reply:
@jlsherrill i believe there is one difference, on line 69, `api.class.client_module(repo.repository_type)`

Reply:
ah i see, i'm fine this for now, i think we could improve it in the future though

Reply:
@jlsherrill @ianballou I'm to fix this, but I'm not sure what the syntax would be for a conditional argument error here. I'm assuming the alternative is to have a conditional for generic in repository.rb instead.